component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java

```diff
@@ -0,0 +1,249 @@
+                                + "queue.manager = 'ESBQManager',"
+                                + "password='1920',"
+                                + "username='mqm',"
+                                + "@map(type='text'))"
+                                + "define stream SweetProductionStream(name string, amount double);"),
+        }
+)
+
+public class IBMMQSink extends Sink {
+    private static final Logger LOG = LoggerFactory.getLogger(IBMMQSink.class);
+    private OptionHolder optionHolder;
+    private QueueConnection connection;
+    private MQQueueConnectionFactory connectionFactory;
+    private StreamDefinition outputStreamDefinition;
+    private QueueSession session;
+    private Queue queue;
+    private QueueSender messageSender;
+    private MessageConsumer consumer;
+    private String userName;
+    private String password;
+    private String queueName;
+    private boolean isSecured = false;
+    private SiddhiAppContext siddhiAppContext;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+        return new Class[]{String.class, Map.class, ByteBuffer.class};
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+        return new String[0];
+    }
+
+    @Override
+    protected void init(StreamDefinition outputStreamDefinition, OptionHolder optionHolder, ConfigReader
+            sinkConfigReader, SiddhiAppContext siddhiAppContext) {
+        this.siddhiAppContext = siddhiAppContext;
+        this.optionHolder = optionHolder;
+        this.connectionFactory = new MQQueueConnectionFactory();
+        this.queueName = optionHolder.validateAndGetStaticValue(IBMMQConstants.DESTINATION_NAME);
+        this.userName = optionHolder.validateAndGetStaticValue(IBMMQConstants.USER_NAME,
+                sinkConfigReader.readConfig(IBMMQConstants.USER_NAME, null));
+        this.password = optionHolder.validateAndGetStaticValue(IBMMQConstants.PASSWORD,
+                sinkConfigReader.readConfig(IBMMQConstants.PASSWORD, null));
+
+        if (Objects.nonNull(userName) && Objects.nonNull(password)) {
+            isSecured = true;
+        }
+        try {
+            connectionFactory.setQueueManager(optionHolder.validateAndGetOption(IBMMQConstants.QUEUE_MANAGER_NAME).
+                    getValue());
+            connectionFactory.setTransportType(WMQConstants.WMQ_CM_CLIENT);
+            connectionFactory.setPort(Integer.parseInt(optionHolder.validateAndGetOption(IBMMQConstants.PORT)
+                    .getValue()));
+            connectionFactory.setHostName(optionHolder.validateAndGetOption(IBMMQConstants.HOST).getValue());
+            connectionFactory.setChannel(optionHolder.validateAndGetOption(IBMMQConstants.CHANNEL).getValue());
+        } catch (JMSException e) {
+            throw new IBMMQSinkAdaptorRuntimeException("Error while initializing IBM MQ sink: " + optionHolder.
+                    validateAndGetOption(IBMMQConstants.DESTINATION_NAME).getValue() +
```

Comment:
Please check whether we can get this into two lines.

Reply:
Fixed by 71dad9c6b3c676a5a94d62bcb9ce7fcd5b91e918